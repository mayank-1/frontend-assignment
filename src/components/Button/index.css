button {
  padding: 0.5rem 1rem;
  border: 1px solid #e0e0e0;
  background: white;
  cursor: pointer;
  border-radius: 4px;
  transition: all 0.2s ease;
  display: flex;
  justify-content: center;
  align-items: center;

  & > svg:first-child {
    margin-right: 5px;
    height: 10px;
  }

  & > svg:last-child {
    margin-left: 5px;
    height: 10px;
  }

  &:hover:not(:disabled) {
    background: #f5f6fa;
    border-color: #3498db;
    color: #6f7373;
  }

  &:disabled {
    opacity: 0.5;
    cursor: not-allowed;
  }

  &.active {
    background: #3498db;
    color: white;
    border-color: #3498db;
  }
}