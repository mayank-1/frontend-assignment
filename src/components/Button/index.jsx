import PropTypes from "prop-types";

// CSS
import "./index.css";

const Button = ({
  onClick,
  disabled = false,
  active = false,
  children,
  label = "",
}) => {
  return (
    <button
      onClick={onClick}
      disabled={disabled}
      className={active ? "active" : ""}
      aria-label={label}
      aria-current={active ? "page" : undefined}
    >
      {children}
    </button>
  );
};

Button.propTypes = {
  onClick: PropTypes.func.isRequired,
  disabled: PropTypes.bool,
  active: PropTypes.bool,
  children: PropTypes.node.isRequired,
  label: PropTypes.string,
};

export default Button;
