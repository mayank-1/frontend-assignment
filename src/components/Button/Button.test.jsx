import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { describe, it, expect } from "vitest";
import Button from "./index"; // Path to your Button component

describe("Button", () => {
  it("renders correctly with default props", () => {
    render(<Button onClick={() => {}} children="Click me" />);
    expect(screen.getByRole("button")).toMatchSnapshot();
  });

  it("renders correctly with disabled prop", () => {
    render(<Button onClick={() => {}} disabled children="Click me" />);
    expect(screen.getByRole("button")).toBeDisabled();
    expect(screen.getByRole("button")).toMatchSnapshot();
  });

  it("renders correctly with active prop", () => {
    render(<Button onClick={() => {}} active children="Click me" />);
    expect(screen.getByRole("button")).toHaveClass("active");
    expect(screen.getByRole("button")).toHaveAttribute("aria-current", "page");
    expect(screen.getByRole("button")).toMatchSnapshot();
  });

  it("renders correctly with label prop", () => {
    render(
      <Button onClick={() => {}} label="Submit Form" children="Click me" />
    );
    expect(screen.getByRole("button")).toHaveAttribute(
      "aria-label",
      "Submit Form"
    );
    expect(screen.getByRole("button")).toMatchSnapshot();
  });

  it("calls onClick when clicked", async () => {
    const handleClick = vi.fn();
    render(<Button onClick={handleClick} children="Click me" />);
    const user = userEvent.setup();
    await user.click(screen.getByRole("button"));
    expect(handleClick).toHaveBeenCalledTimes(1);
  });

  it("does not call onClick when disabled", async () => {
    const handleClick = vi.fn();
    render(<Button onClick={handleClick} disabled children="Click me" />);
    const user = userEvent.setup();
    await user.click(screen.getByRole("button"));
    expect(handleClick).not.toHaveBeenCalled();
  });
});
