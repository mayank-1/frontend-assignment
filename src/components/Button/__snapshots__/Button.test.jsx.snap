// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button > renders correctly with active prop 1`] = `
<button
  aria-current="page"
  aria-label=""
  class="active"
>
  Click me
</button>
`;

exports[`Button > renders correctly with default props 1`] = `
<button
  aria-label=""
  class=""
>
  Click me
</button>
`;

exports[`Button > renders correctly with disabled prop 1`] = `
<button
  aria-label=""
  class=""
  disabled=""
>
  Click me
</button>
`;

exports[`Button > renders correctly with label prop 1`] = `
<button
  aria-label="Submit Form"
  class=""
>
  Click me
</button>
`;
